### PR TITLE
Feature/likes manager

### PR DESCRIPTION
**Pull request by benjivou**

### CODE REVIEW

File: app/src/main/java/com/example/movieapp/likesmanager/AppDatabase.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.example.movieapp.likesmanager
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/likesmanager/AppDatabase.kt]: It is the mess in your packages =D

File: app/src/main/java/com/example/movieapp/MovieViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        binding.likeBtn.setOnClickListener(LikeClicker(movie, binding))
+    }
+
+    class LikeClicker(val movie: Movie, private val binding: ListItemBinding) :
+        View.OnClickListener {
+
+        init {
+            MainScope().launch(Dispatchers.IO) {
+                if (App.database.movieDAO().isLiked(movie.id)) {
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: It is not the ViewHolder job to do this kind of job -> move it to you viewModel

File: app/src/main/java/com/example/movieapp/MovieViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    getString(R.string.itemRate, movie.voteAverage.toString())
+
+                Picasso.get().load(PURL + movie.posterPath)
+                    .into(image)
+            }
+        }
+        binding.likeBtn.setOnClickListener(LikeClicker(movie, binding))
+    }
+
+    class LikeClicker(val movie: Movie, private val binding: ListItemBinding) :
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: useless

File: app/src/main/java/com/example/movieapp/crawler/pojo/Movie.kt [hunk 4]
```diff
@@ -1,24 +1,98 @@
...
+    @ColumnInfo(name = "vote_average") @SerializedName("vote_average") val voteAverage: Float,
+    @ColumnInfo(name = "overview") @SerializedName("overview") val overview: String,
+    @ColumnInfo(name = "release_date") @SerializedName("release_date") val releaseDate: String,
+    @ColumnInfo(
+        name = "image",
+        typeAffinity = ColumnInfo.BLOB
+    ) @SerializedName("image") var image: ByteArray? = null
+
+) {
+    override fun equals(other: Any?): Boolean {
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/crawler/pojo/Movie.kt]: why ?

File: app/src/main/java/com/example/movieapp/crawler/pojo/Movie.kt [hunk 5]
```diff
@@ -1,24 +1,98 @@
...
+        if (releaseDate != other.releaseDate) return false
+        if (image != null) {
+            if (other.image == null) return false
+            if (!image!!.contentEquals(other.image!!)) return false
+        } else if (other.image != null) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/crawler/pojo/Movie.kt]: why ?

**Review by alba-benoit — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/movieapp/MovieViewHolder.kt (hunk 3 above)
Comment (benjivou) [reply app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: Quelle ligne ?

File: app/src/main/java/com/example/movieapp/likesmanager/AppDatabase.kt (hunk 1 above)
Comment (benjivou) [reply app/src/main/java/com/example/movieapp/likesmanager/AppDatabase.kt]: je l'ai fait dans la nouvelle branche

File: app/src/main/java/com/example/movieapp/MovieViewHolder.kt (hunk 2 above)
Comment (benjivou) [reply app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: The Database Call ?

Comment (benjivou) [reply app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: Or just the init ?

File: app/src/main/java/com/example/movieapp/MovieViewHolder.kt (hunk 3 above)
Comment (benjivou) [reply app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: the class is useless or the binding ?

Comment (alba-benoit) [reply app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: the class

File: app/src/main/java/com/example/movieapp/MovieViewHolder.kt (hunk 2 above)
Comment (benjivou) [reply app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: Done

File: app/src/main/java/com/example/movieapp/MovieViewHolder.kt (hunk 3 above)
Comment (benjivou) [reply app/src/main/java/com/example/movieapp/MovieViewHolder.kt]: Done

File: app/src/main/java/com/example/movieapp/view/MovieViewHolder.kt [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                    getString(R.string.itemRate, movie!!.voteAverage.toString())
+
+                Picasso.get().load(PURL + movie!!.posterPath)
+                    .into(image)
+            }
+        }
+        binding.likeBtn.setOnClickListener(
+            if (pair.second)
+                View.OnClickListener {
+                    mainViewModel.deleteMovie(movie!!)
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/view/MovieViewHolder.kt]: do not call viewModel method here

**Review by alba-benoit — CHANGES_REQUESTED** [review]

**Pull request merged**